examples/rn-todo-list/README.md

```diff
@@ -0,0 +1,208 @@
+  * Shows configuration for opening a synced Realm.
+    * Realms are opened immediately without waiting for downloads from the server.
+    * See [Offline Support](#note-offline-support) below.
+* Allowing users to only read and write to their own tasks via [data access rules/permissions](https://www.mongodb.com/docs/atlas/app-services/rules/roles/#define-roles---permissions).
+  * See [Set Data Access Permissions](#set-data-access-permissions) further below.
+* The app demonstrates both how to use *either*:
+  * A) A local-only app (see [AppNonSync.tsx](./frontend/app/AppNonSync.tsx)).
+  * B) A [Device Sync](https://www.mongodb.com/atlas/app-services/device-sync) enabled app (see [AppSync.tsx](./frontend/app/AppSync.tsx)).
+  * Choose which one to use via [sync.config.js](./frontend/sync.config.js).
+
+### Note: Offline Support
+
+Users who have logged in at least once will have their credentials cached on the client. Thus, a logged in user who restarts the app will remain logged in. [@realm/react's](https://www.npmjs.com/package/@realm/react) `UserProvider` automatically handles this for you by checking if the `app.currentUser` already exists.
+
+Data that was previously synced to the device will also exist locally in the Realm database. From this point on, users can be offline and still query and update data. Any changes made offline will be synced automatically to Atlas and any other devices once a network connection is established. If multiple users modify the same data either while online or offline, those conflicts are [automatically resolved](https://www.mongodb.com/docs/atlas/app-services/sync/details/conflict-resolution/) before being synced.
+
+#### Realm Configuration
+
+When opening a Realm, we can specify the behavior in the Realm configuration when opening it for the first time (via `newRealmFileBehavior`) and for subsequent ones (via `existingRealmFileBehavior`). We can either:
+* `OpenRealmBehaviorType.OpenImmediately`
+  * Opens the Realm file immediately if it exists, otherwise it first creates a new empty Realm file then opens it.
+  * This lets users use the app with the existing data, while syncing any changes to the device in the background.
+* `OpenRealmBehaviorType.DownloadBeforeOpen`
+  * If there is data to be downloaded, this waits for the data to be fully synced before opening the Realm.
+
+This app opens a Realm via `RealmProvider` (see [AppSync.tsx](./frontend/app/AppSync.tsx)) and passes the configuration as props. We use `OpenImmediately` for new and existing Realm files in order to use the app while offline.
+
+> See [OpenRealmBehaviorConfiguration](https://www.mongodb.com/docs/realm-sdks/js/latest/types/OpenRealmBehaviorConfiguration.html) for possible configurations of new and existing Realm file behaviors.
+
+## Getting Started
+
+### Prerequisites
+
+* [Node.js](https://nodejs.org/)
+* [React Native development environment](https://reactnative.dev/docs/environment-setup?guide=native)
```

Comment:
For the React Native prereq here, should we also mention that it uses the React Native CLI Quickstart rather than the Expo one?

Reply:
Yes, otherwise it might cause confusion.